### PR TITLE
Move CheckDatabase() methods from main to servers

### PR DESCRIPTION
**Pull request by codingllama**

### CODE REVIEW

File: server/vmap/trillian_map_server/main.go [hunk 1]
```diff
@@ -110,13 +96,16 @@ func main() {
...
-		glog.Errorf("Failed to listen on the server port: %d, because: %v", *serverPortFlag, err)
-		os.Exit(1)
+		glog.Exitf("Failed to listen on the server port: %d, because: %v", *serverPortFlag, err)
 	}
 
 	// Bring up the RPC server and then block until we get a signal to stop
-	rpcServer := startRPCServer(registry)
+	rpcServer, err := startRPCServer(registry)
+	if err != nil {
+		glog.Errorf("Failed to start RPC server: %v", err)
```
Comment (AlCutter) [inline server/vmap/trillian_map_server/main.go]: Should this be an `Exitf`?

File: server/vmap/trillian_map_server.go [hunk 2]
```diff
@@ -39,6 +39,16 @@ func NewTrillianMapServer(registry extension.Registry) *TrillianMapServer {
 	return &TrillianMapServer{registry}
 }
 
+// CheckDatabaseAccessible returns nil if the database is accessible, error
+// otherwise.
+func (t *TrillianMapServer) CheckDatabaseAccessible() error {
```
Comment (AlCutter) [inline server/vmap/trillian_map_server.go]: How about `BackendsHealthy` or something less specific to databases?

File: server/vmap/trillian_map_server/main.go (hunk 1 above)
Comment (codingllama) [inline server/vmap/trillian_map_server/main.go]: Done (changed trillian_log_server/main.go too).

File: server/vmap/trillian_map_server.go (hunk 2 above)
Comment (codingllama) [inline server/vmap/trillian_map_server.go]: Changed to IsHealthy in both servers.

**Comment by codingllama** [comment]
PTAL?

**Pull request merged**